usr/src/uts/common/brand/lx/syscall/lx_socket.c

```diff
@@ -3951,6 +3984,35 @@ lx_getsockopt_tcp(sonode_t *so, int optname, void *optval, socklen_t *optlen)
 		}
 		*optlen = sizeof (int);
 		goto out;
+
+	case LX_TCP_INFO:
+		/*
+		 * We only try to fill in the fields, that we know
+		 * some applications expect.
+		 */
+		if (*optlen < sizeof (lx_tcp_info_t)) {
+			error = EINVAL;
+		} else {
+			*optlen = sizeof (lx_tcp_info_t);
+			bzero(optval, *optlen);
+
+			if (so->so_type != SOCK_STREAM)
+			   goto out; 
+
+			lx_tcp_info_t *ti = (lx_tcp_info_t *)optval;
+			conn_t *con = (struct conn_s *)so->so_proto_handle;
+			tcp_t *tp = con->conn_tcp;
+			ti->tcpi_state = STOL_TCPSTATE(tp->tcp_state);
+			ti->tcpi_rto = tp->tcp_rto * 1000;/* Round trip timeout */
```

Comment:
Cstyle nit, needs a space...

Actually:

```
/* tcp_rto is kept in msec, but the API wants things in usec. */
ti->tcpi_rto = tp->tcp_rto * 1000;
```

Reply:
It's fixed now and updated the comment.